src/test/groovy/edu/ie3/datamodel/models/input/system/StorageInputTest.groovy

```diff
@@ -0,0 +1,32 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.models.input.system
+
+import edu.ie3.test.common.SystemParticipantTestData
+import spock.lang.Specification
+
+
+class StorageInputTest extends Specification {
+
+	def "A StorageInput copy method should work as expected"() {
+		given:
+		def storageInput = SystemParticipantTestData.storageInput
+
+		when:
+		def alteredUnit = storageInput.copy().type(SystemParticipantTestData.storageTypeInput).build()
```

Comment:
`#behaviour()` is not tested.

Reply:
by purpose because it's deprecated and marked accordingly. There's no sense in writing tests for features that are not supported anymore.